File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/solutionsvb/vb/getsolutiondependencies3.vb
' <snippetgetsolutiondependencies3>


 Dim optionSetMetadata As New OptionSetMetadata() With {
  .Name = _globalOptionSetName,
  .DisplayName = New Label("Example Option Set", _languageCode),
  .IsGlobal = True,
  .OptionSetType = OptionSetType.Picklist
 }
 optionSetMetadata.Options.AddRange(
  {
   New OptionMetadata(New Label("Option 1", _languageCode), 1),
   New OptionMetadata(New Label("Option 2", _languageCode), 2)
  }
 )
 Dim createOptionSetRequest As CreateOptionSetRequest =
  New CreateOptionSetRequest With {
   .OptionSet = optionSetMetadata
  }

 createOptionSetRequest.SolutionUniqueName = _primarySolutionName
 _serviceProxy.Execute(createOptionSetRequest)

' </snippetgetsolutiondependencies3>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/solutions/cs/getsolutiondependencies3.cs
// <snippetgetsolutiondependencies3>


 OptionSetMetadata optionSetMetadata = new OptionSetMetadata()
 {
     Name = _globalOptionSetName,
     DisplayName = new Label("Example Option Set", _languageCode),
     IsGlobal = true,
     OptionSetType = OptionSetType.Picklist,
     Options =
 {
     new OptionMetadata(new Label("Option 1", _languageCode), 1),
     new OptionMetadata(new Label("Option 2", _languageCode), 2)
 }
 };
 CreateOptionSetRequest createOptionSetRequest = new CreateOptionSetRequest
 {
     OptionSet = optionSetMetadata                
 };

 createOptionSetRequest.SolutionUniqueName = _primarySolutionName;
 _serviceProxy.Execute(createOptionSetRequest);

// </snippetgetsolutiondependencies3>